crowbar_framework/app/controllers/barclamp_controller.rb

```diff
@@ -116,20 +116,27 @@ def transition
     state = params[:state] # State of node transitioning
     name = params[:name] # Name of node transitioning
 
-    unless valid_transition_states.include?(state)
-      render text: "State '#{state}' is not valid.", status: 400
+    node = Node.find_node_by_name_or_alias(name)
+
+    if node.nil?
+      render text: "Node does not exist.", status: 404
     else
-      status, response = @service_object.transition(id, name, state)
-      if status != 200
-        render text: response, status: status
-      else
-        # Be backward compatible with barclamps returning a node hash, passing
-        # them intact.
-        if response[:name]
-          render json: Node.find_by_name(response[:name]).to_hash
+      name = node.name
```

Comment:
Why not simply pass `node.name` to `transition()` below? The variable is not used anywhere else.